brainframe_qt/ui/dialogs/task_configuration/core/zone.py

```diff
@@ -0,0 +1,139 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import List, Optional, Tuple, TypeVar
+
+from brainframe.api import bf_codecs
+from shapely import geometry
+
+T = TypeVar("T")
+
+
+@dataclass
+class Zone(ABC):
```

Comment:
Why not just subclass from the Zone codec dataclass, instead of re-specifying and re-documenting identical fields?